Added BanModule

asapverneri opened the pull request:
Simple ready ban module with custom reason using cvar 'css_ban'

asapverneri committed:
Added BanModule

Simple ban module with custom reason using cvar 'css_ban'

asapverneri committed:
minor fix for simple ban module

asapverneri committed:
Update Plugin.cs

asapverneri committed:
Update Plugin.cs

asapverneri committed:
little update

KillStr3aK commented:
Hey!
Can you add the build directory to the `.gitignore` file?

asapverneri committed:
Added to .gitignore

KillStr3aK commented on .gitignore:
```diff
@@ -360,4 +360,5 @@ MigrationBackup/
-FodyWeavers.xsd
\ No newline at end of file
+FodyWeavers.xsd
+/src/AntiDLL.BanModule/build
```
I believe a simple `build/` would be better as that also handles any folder named `build`, just in case there would be more projects with the same output folder

KillStr3aK commented on src/AntiDLL.BanModule/Plugin.cs:
```diff
@@ -0,0 +1,88 @@
+                AddTimer(10.0f, () => Punishedplayers.Remove(player.SteamID));
+            }
+
+            
```
needless gap

KillStr3aK commented on src/AntiDLL.BanModule/Plugin.cs:
```diff
@@ -0,0 +1,88 @@
+
+        private void OnDetection(CCSPlayerController player, string eventName)
+        {
+
```
needless gap

KillStr3aK commented on src/AntiDLL.BanModule/Plugin.cs:
```diff
@@ -0,0 +1,88 @@
+
+        public override string ModuleAuthor => "verneri";
+
+        private static PluginCapability<IAntiDLL> AntiDLL { get; } = new PluginCapability<IAntiDLL>("AntiDLL");
+        public Config Config { get; set; } = new();
+        private HashSet<ulong> Punishedplayers = [];
```
Could you add an empty line between the properties/fields?, just like above

KillStr3aK commented on src/AntiDLL.BanModule/Plugin.cs:
```diff
@@ -0,0 +1,88 @@
+                Punishedplayers.Add(player.SteamID);
+                Logger.LogInformation($"Added {player.PlayerName} to 'Punishedplayers' list.");
+
+
```
needless gap

KillStr3aK reviewed (changes requested):
I don't want to look very picky, just lets keep this stuff clean

asapverneri committed:
minor changes

asapverneri commented:
> I don't want to look very picky, just lets keep this stuff clean

no worries, made these changes. Let me know if anything else

GabrielBigardi committed:
Fix wrong variable name

The user may wan't to kick instead of banning, so using the name "BanReason" can be confusing.

asapverneri committed:
Merge pull request #1 from GabrielBigardi/master

Fix wrong variable name

KillStr3aK committed:
chore: codestyle changes

KillStr3aK commented:
Thanks for your contribution!
